cabal-install/src/Distribution/Client/ProjectBuilding/UnpackedPackage.hs

```diff
@@ -678,7 +679,32 @@ buildAndInstallUnpackedPackage
           runConfigure
         PBBuildPhase{runBuild} -> do
           noticeProgress ProgressBuilding
-          runBuild
+          hookDir <- (</> ".cabal/hooks") <$> getEnv "HOME"
+          -- run preBuildHook. If it returns with 0, we assume the build was
+          -- successful. If not, run the build.
+          code <-
+            rawSystemExitCode
+              verbosity
+              (Just srcdir)
+              (hookDir </> "preBuildHook")
+              [ (unUnitId $ installedUnitId rpkg)
+              , (getSymbolicPath srcdir)
+              , (getSymbolicPath builddir)
+              ]
+              `catchIO` (\_ -> return (ExitFailure 10))
```

Comment:
I'm not convinced we should overwrite the exit code of the script?

Reply:
there is no exit code. This is `file not found` mapped to `ExitFailure 10`. Yes 10 is just a magic constant.